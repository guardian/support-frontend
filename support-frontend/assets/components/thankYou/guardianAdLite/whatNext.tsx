import { palette } from '@guardian/source/foundations';
import OrderedList from 'components/list/orderedList';
import type { ObserverPrint } from 'pages/[countryGroupId]/components/thankYouComponent';
import BulletPointedList from '../utilityComponents/BulletPointedList';

type WhatNextProps = {
	amount: string;
	startDate?: string;
	isSignedIn?: boolean;
	observerPrint?: ObserverPrint;
};

export function WhatNext({
	amount,
	startDate,
	isSignedIn = false,
	observerPrint,
}: WhatNextProps): JSX.Element {
	const whatNextItems = [
		!observerPrint
			? 'You will receive an email confirming the details of your subscription'
			: '',
		!observerPrint && !!startDate
			? `Your payment of Â£${amount}/month will be taken on ${startDate}`
			: '',
		!observerPrint && isSignedIn
			? 'You can now start reading the Guardian website on all your devices without personalised advertising'
			: '',
		observerPrint
			? 'Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.'
			: '',
		observerPrint ? `Your newspaper will be delivered to your door.` : '',
		observerPrint === 'ObserverSubscriptionCard'
			? 'Visit your chosen participating newsagent to pick up your newspaper using your Subscription Card, or arrange a home delivery using your delivery letter.'
			: '',
	];

	const bulletItems: string[] = whatNextItems.filter((item) => item);

	return (
		<>
			{observerPrint ? (
				<OrderedList items={bulletItems} />
			) : (
				<BulletPointedList items={bulletItems} color={palette.neutral[7]} />
			)}
		</>
	);
}
