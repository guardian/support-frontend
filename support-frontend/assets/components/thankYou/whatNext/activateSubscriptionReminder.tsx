import { css } from '@emotion/react';
import { from, space } from '@guardian/source/foundations';

const paragraphCheckInbox = css`
	font-weight: bold;
	margin-top: ${space[5]}px;
	${from.tablet} {
		margin-top: ${space[6]}px;
	}
`;

export function ActivateSubscriptionReminder(): JSX.Element {
	return (
		<>
			<p>
				While your subscription is ready to use on this device, you will need to
				finish setting up your account to read the Guardian website with
				non-personalised advertising on other devices or browsers. You will also
				be able to manage your subscription.
			</p>
			<p css={paragraphCheckInbox}>
				Check your inbox, find the email with the subject line ‘Complete your
				Guardian account’, and follow the link to set your password.
			</p>
		</>
	);
}
