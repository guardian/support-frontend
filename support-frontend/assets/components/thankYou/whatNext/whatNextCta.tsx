import { css } from '@emotion/react';
import { palette, space } from '@guardian/source/foundations';
import {
	LinkButton,
	SvgArrowRightStraight,
} from '@guardian/source/react-components';
import { sendTrackingEventsOnClick } from 'helpers/productPrice/subscriptions';
import { helpCentreUrl, manageSubsUrl } from 'helpers/urls/externalLinks';

const btnStyleOverrides = css`
	justify-content: left;
	margin-bottom: ${space[5]}px;
`;

const neutralFontColor = css`
	color: ${palette.neutral[0]};
`;

type AddressCtaProp = {
	copy: string;
	address: string;
	hasArrow?: boolean;
};

export function AddressCta({
	address,
	copy,
	hasArrow = false,
}: AddressCtaProp): JSX.Element {
	return (
		<>
			{hasArrow ? (
				<LinkButton
					cssOverrides={btnStyleOverrides}
					href={address}
					size="default"
					icon={<SvgArrowRightStraight />}
					iconSide="right"
					nudgeIcon
					aria-label={copy}
				>
					{copy}
				</LinkButton>
			) : (
				<LinkButton
					cssOverrides={btnStyleOverrides}
					href={address}
					size="default"
					aria-label={copy}
				>
					{copy}
				</LinkButton>
			)}
		</>
	);
}

export const myAccountCta = (
	<a
		css={neutralFontColor}
		href={manageSubsUrl}
		onClick={sendTrackingEventsOnClick({
			id: 'checkout_my_account',
			product: 'Paper',
			componentType: 'ACQUISITIONS_BUTTON',
		})}
	>
		Manage my account
	</a>
);

export const helpCenterCta = (linkCopy: string) => (
	<a
		css={neutralFontColor}
		href={helpCentreUrl}
		onClick={sendTrackingEventsOnClick({
			id: 'checkout_help_centre',
			product: 'Paper',
			componentType: 'ACQUISITIONS_BUTTON',
		})}
	>
		{linkCopy}
	</a>
);
