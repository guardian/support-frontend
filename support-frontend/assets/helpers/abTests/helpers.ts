import type { CountryGroupId } from '@modules/internationalisation/countryGroup';
import type {
	AmountsTest,
	AmountsVariant,
	SelectedAmountsVariant,
} from 'helpers/contributions';

type AmountsTestWithVariants = AmountsTest & {
	variants: [AmountsVariant, ...AmountsVariant[]];
};

type NonEmptyAmountsTestArray = [
	AmountsTestWithVariants,
	...AmountsTestWithVariants[],
];

const FALLBACK_AMOUNTS: NonEmptyAmountsTestArray = [
	{
		testName: 'FALLBACK_AMOUNTS__GBPCountries',
		liveTestName: '',
		isLive: false,
		targeting: {
			targetingType: 'Region',
			region: 'GBPCountries',
		},
		order: 0,
		seed: 0,
		variants: [
			{
				variantName: 'CONTROL',
				defaultContributionType: 'MONTHLY',
				displayContributionType: ['ONE_OFF', 'MONTHLY', 'ANNUAL'],
				amountsCardData: {
					ONE_OFF: {
						amounts: [30, 60, 120, 240],
						defaultAmount: 60,
						hideChooseYourAmount: false,
					},
					MONTHLY: {
						amounts: [3, 7, 12],
						defaultAmount: 7,
						hideChooseYourAmount: false,
					},
					ANNUAL: {
						amounts: [60, 120, 240, 480],
						defaultAmount: 120,
						hideChooseYourAmount: false,
					},
				},
			},
		],
	},
	{
		testName: 'FALLBACK_AMOUNTS__UnitedStates',
		liveTestName: '',
		isLive: false,
		targeting: {
			targetingType: 'Region',
			region: 'UnitedStates',
		},
		order: 0,
		seed: 0,
		variants: [
			{
				variantName: 'CONTROL',
				defaultContributionType: 'MONTHLY',
				displayContributionType: ['ONE_OFF', 'MONTHLY', 'ANNUAL'],
				amountsCardData: {
					ONE_OFF: {
						amounts: [25, 50, 100, 250],
						defaultAmount: 50,
						hideChooseYourAmount: false,
					},
					MONTHLY: {
						amounts: [7, 15, 30],
						defaultAmount: 15,
						hideChooseYourAmount: false,
					},
					ANNUAL: {
						amounts: [50, 100, 250, 500],
						defaultAmount: 50,
						hideChooseYourAmount: false,
					},
				},
			},
		],
	},
	{
		testName: 'FALLBACK_AMOUNTS__EURCountries',
		liveTestName: '',
		isLive: false,
		targeting: {
			targetingType: 'Region',
			region: 'EURCountries',
		},
		order: 0,
		seed: 0,
		variants: [
			{
				variantName: 'CONTROL',
				defaultContributionType: 'MONTHLY',
				displayContributionType: ['ONE_OFF', 'MONTHLY', 'ANNUAL'],
				amountsCardData: {
					ONE_OFF: {
						amounts: [25, 50, 100, 250],
						defaultAmount: 50,
						hideChooseYourAmount: false,
					},
					MONTHLY: {
						amounts: [6, 10, 20],
						defaultAmount: 10,
						hideChooseYourAmount: false,
					},
					ANNUAL: {
						amounts: [50, 100, 250, 500],
						defaultAmount: 50,
						hideChooseYourAmount: false,
					},
				},
			},
		],
	},
	{
		testName: 'FALLBACK_AMOUNTS__International',
		liveTestName: '',
		isLive: false,
		targeting: {
			targetingType: 'Region',
			region: 'International',
		},
		order: 0,
		seed: 0,
		variants: [
			{
				variantName: 'CONTROL',
				defaultContributionType: 'MONTHLY',
				displayContributionType: ['ONE_OFF', 'MONTHLY', 'ANNUAL'],
				amountsCardData: {
					ONE_OFF: {
						amounts: [25, 50, 100, 250],
						defaultAmount: 50,
						hideChooseYourAmount: false,
					},
					MONTHLY: {
						amounts: [5, 10, 20],
						defaultAmount: 10,
						hideChooseYourAmount: false,
					},
					ANNUAL: {
						amounts: [60, 100, 250, 500],
						defaultAmount: 60,
						hideChooseYourAmount: false,
					},
				},
			},
		],
	},
	{
		testName: 'FALLBACK_AMOUNTS__Canada',
		liveTestName: '',
		isLive: false,
		targeting: {
			targetingType: 'Region',
			region: 'Canada',
		},
		order: 0,
		seed: 0,
		variants: [
			{
				variantName: 'CONTROL',
				defaultContributionType: 'MONTHLY',
				displayContributionType: ['ONE_OFF', 'MONTHLY', 'ANNUAL'],
				amountsCardData: {
					ONE_OFF: {
						amounts: [25, 50, 100, 250],
						defaultAmount: 50,
						hideChooseYourAmount: false,
					},
					MONTHLY: {
						amounts: [5, 10, 20],
						defaultAmount: 10,
						hideChooseYourAmount: false,
					},
					ANNUAL: {
						amounts: [60, 100, 250, 500],
						defaultAmount: 60,
						hideChooseYourAmount: false,
					},
				},
			},
		],
	},
	{
		testName: 'FALLBACK_AMOUNTS__AUDCountries',
		liveTestName: '',
		isLive: false,
		targeting: {
			targetingType: 'Region',
			region: 'AUDCountries',
		},
		order: 0,
		seed: 0,
		variants: [
			{
				variantName: 'CONTROL',
				defaultContributionType: 'MONTHLY',
				displayContributionType: ['ONE_OFF', 'MONTHLY', 'ANNUAL'],
				amountsCardData: {
					ONE_OFF: {
						amounts: [60, 100, 250, 500],
						defaultAmount: 100,
						hideChooseYourAmount: false,
					},
					MONTHLY: {
						amounts: [10, 20, 40],
						defaultAmount: 20,
						hideChooseYourAmount: false,
					},
					ANNUAL: {
						amounts: [80, 250, 500, 750],
						defaultAmount: 80,
						hideChooseYourAmount: false,
					},
				},
			},
		],
	},
	{
		testName: 'FALLBACK_AMOUNTS__NZDCountries',
		liveTestName: '',
		isLive: false,
		targeting: {
			targetingType: 'Region',
			region: 'NZDCountries',
		},
		order: 0,
		seed: 0,
		variants: [
			{
				variantName: 'CONTROL',
				defaultContributionType: 'MONTHLY',
				displayContributionType: ['ONE_OFF', 'MONTHLY', 'ANNUAL'],
				amountsCardData: {
					ONE_OFF: {
						amounts: [50, 100, 250, 500],
						defaultAmount: 100,
						hideChooseYourAmount: false,
					},
					MONTHLY: {
						amounts: [10, 20, 50],
						defaultAmount: 20,
						hideChooseYourAmount: false,
					},
					ANNUAL: {
						amounts: [50, 100, 250, 500],
						defaultAmount: 50,
						hideChooseYourAmount: false,
					},
				},
			},
		],
	},
];

export function getFallbackAmounts(
	countryGroupId: CountryGroupId,
): SelectedAmountsVariant {
	// Create fallback data - the amounts card must always have data to dsplay
	const fallbackTest = FALLBACK_AMOUNTS.find(
		(t) =>
			t.targeting.targetingType === 'Region' &&
			t.targeting.region === countryGroupId,
	);
	if (fallbackTest) {
		return {
			...fallbackTest.variants[0],
			testName: fallbackTest.testName,
		};
	} else {
		return {
			...FALLBACK_AMOUNTS[0].variants[0],
			testName: FALLBACK_AMOUNTS[0].testName,
		};
	}
}
