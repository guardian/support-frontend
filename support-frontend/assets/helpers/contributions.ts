// ----- Imports ----- //
import type { IsoCountry } from 'helpers/internationalisation/country';
import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
import { roundToDecimalPlaces } from 'helpers/utilities/utilities';

// ----- Types ----- //
export const contributionTypes = ['ONE_OFF', 'MONTHLY', 'ANNUAL'];

type RegularContributionTypeMap<T> = {
	MONTHLY: T;
	ANNUAL: T;
};

type ContributionTypeMap<T> = RegularContributionTypeMap<T> & {
	ONE_OFF: T;
};

export type ContributionType = keyof ContributionTypeMap<null>;

export interface AmountValuesObject {
	amounts: number[];
	defaultAmount: number;
	hideChooseYourAmount: boolean;
}

type AmountsCardData = {
	[key in ContributionType]: AmountValuesObject;
};

export interface AmountsVariant {
	variantName: string;
	defaultContributionType: ContributionType;
	displayContributionType: ContributionType[];
	amountsCardData: AmountsCardData;
}

export type AmountsTestTargeting =
	| { targetingType: 'Region'; region: CountryGroupId }
	| { targetingType: 'Country'; countries: IsoCountry[] };

export interface AmountsTest {
	testName: string;
	liveTestName?: string;
	isLive: boolean;
	targeting: AmountsTestTargeting;
	order: number;
	seed: number;
	variants: AmountsVariant[];
}

export type AmountsTests = AmountsTest[];

export interface SelectedAmountsVariant extends AmountsVariant {
	testName: string;
}

type ContributionTypeSetting = {
	contributionType: ContributionType;
	isDefault?: boolean;
};

export type ContributionTypes = Record<
	CountryGroupId,
	ContributionTypeSetting[]
>;

type Config = Record<
	ContributionType,
	{
		min: number;
		minInWords: string;
		max: number;
		maxInWords: string;
		default: number; // TODO - remove this field once old payment flow has gone
	}
>;

export type OtherAmounts = Record<
	ContributionType,
	{
		amount: string | null;
	}
>;

export type SelectedAmounts = Record<ContributionType, number | 'other'>;

const getAmount = (
	selectedAmounts: SelectedAmounts,
	otherAmounts: OtherAmounts,
	contributionType: ContributionType,
	coverTransactionCostSelected?: boolean,
): number => {
	const selectedChoiceCardAmount = selectedAmounts[contributionType];
	/**
	 * TODO: otherAmount falls back to zero if no other amount
	 * entered by user. This prevents the function returning NaN.
	 * Ideally it would return undefined and we'd handle that, but the
	 * impact of doing so was deemed too great, considering most use cases
	 * are in the soon to be deprecated 2-step checkout.
	 */
	const otherAmount = otherAmounts[contributionType].amount ?? '0';

	const selectedAmount =
		selectedChoiceCardAmount === 'other'
			? parseFloat(otherAmount)
			: selectedChoiceCardAmount;

	// Only cover transaction costs for one off contributions
	const coverTransactionCost =
		coverTransactionCostSelected && contributionType === 'ONE_OFF';

	return coverTransactionCost
		? roundToDecimalPlaces(selectedAmount * 1.04)
		: selectedAmount;
};

// ----- Setup ----- //

const numbersInWords = {
	'1': 'one',
	'2': 'two',
	'5': 'five',
	'6': 'six',
	'7': 'seven',
	'10': 'ten',
	'15': 'fifteen',
	'20': 'twenty',
	'25': 'twenty five',
	'30': 'thirty',
	'35': 'thirty five',
	'40': 'forty',
	'50': 'fifty',
	'75': 'seventy five',
	'100': 'one hundred',
	'125': 'one hundred and twenty five',
	'166': 'one hundred and sixty six',
	'200': 'two hundred',
	'250': 'two hundred and fifty',
	'275': 'two hundred and seventy five',
	'500': 'five hundred',
	'750': 'seven hundred and fifty',
	'800': 'eight hundred',
	'2000': 'two thousand',
	'10000': 'ten thousand',
	'16000': 'sixteen thousand',
} as const;

const defaultConfig: Config = {
	ANNUAL: {
		min: 10,
		minInWords: numbersInWords['10'],
		max: 2000,
		maxInWords: numbersInWords['2000'],
		default: 50,
	},
	MONTHLY: {
		min: 2,
		minInWords: numbersInWords['2'],
		max: 166,
		maxInWords: numbersInWords['166'],
		default: 5,
	},
	ONE_OFF: {
		min: 1,
		minInWords: numbersInWords['1'],
		max: 2000,
		maxInWords: numbersInWords['2000'],
		default: 50,
	},
};

const config: Record<CountryGroupId, Config> = {
	GBPCountries: defaultConfig,
	AUDCountries: {
		ANNUAL: defaultConfig.ANNUAL,
		MONTHLY: {
			min: 2,
			minInWords: numbersInWords['2'],
			max: 200,
			maxInWords: numbersInWords['200'],
			default: 20,
		},
		ONE_OFF: {
			min: 1,
			minInWords: numbersInWords['1'],
			max: 16000,
			maxInWords: numbersInWords['16000'],
			default: 50,
		},
	},
	EURCountries: {
		ANNUAL: defaultConfig.ANNUAL,
		MONTHLY: {
			min: 2,
			minInWords: numbersInWords['2'],
			max: 166,
			maxInWords: numbersInWords['166'],
			default: 10,
		},
		ONE_OFF: defaultConfig.ONE_OFF,
	},
	UnitedStates: {
		ANNUAL: {
			...defaultConfig.ANNUAL,
			max: 10000,
			maxInWords: numbersInWords['10000'],
		},
		MONTHLY: {
			min: 2,
			minInWords: numbersInWords['2'],
			max: 800,
			maxInWords: numbersInWords['800'],
			default: 15,
		},
		ONE_OFF: {
			...defaultConfig.ONE_OFF,
			max: 10000,
			maxInWords: numbersInWords['10000'],
		},
	},
	International: {
		ANNUAL: defaultConfig.ANNUAL,
		MONTHLY: {
			min: 2,
			minInWords: numbersInWords['2'],
			max: 166,
			maxInWords: numbersInWords['166'],
			default: 10,
		},
		ONE_OFF: defaultConfig.ONE_OFF,
	},
	NZDCountries: {
		ANNUAL: defaultConfig.ANNUAL,
		MONTHLY: {
			min: 2,
			minInWords: numbersInWords['2'],
			max: 200,
			maxInWords: numbersInWords['200'],
			default: 20,
		},
		ONE_OFF: defaultConfig.ONE_OFF,
	},
	Canada: {
		ANNUAL: defaultConfig.ANNUAL,
		MONTHLY: {
			min: 2,
			minInWords: numbersInWords['2'],
			max: 166,
			maxInWords: numbersInWords['166'],
			default: 10,
		},
		ONE_OFF: defaultConfig.ONE_OFF,
	},
};

const contributionsOnlyAmountsTestName = 'VAT_COMPLIANCE';

const isContributionsOnlyCountry = (
	amountsVariant: SelectedAmountsVariant,
): boolean => {
	const { testName } = amountsVariant;
	return testName === contributionsOnlyAmountsTestName;
};

// ----- Exports ----- //
export {
	config,
	getAmount,
	contributionsOnlyAmountsTestName,
	isContributionsOnlyCountry,
};
