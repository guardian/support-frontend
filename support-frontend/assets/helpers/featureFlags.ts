type FeatureFlag = {
	enablePremiumDigital: boolean;
};

export function getFeatureFlags(): FeatureFlag {
	const urlParams = new URLSearchParams(window.location.search);

	return {
		enablePremiumDigital: urlParams.has('enablePremiumDigital'),
	};
}
