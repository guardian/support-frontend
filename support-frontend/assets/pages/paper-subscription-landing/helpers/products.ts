import type {
	PaperProductOptions,
	ProductOptions,
} from 'helpers/productPrice/productOptions';
import {
	Everyday,
	EverydayPlus,
	SaturdayPlus,
	Sixday,
	SixdayPlus,
	SundayPlus,
	WeekendPlus,
} from 'helpers/productPrice/productOptions';

export const getTitle = (productOption: ProductOptions) => {
	switch (productOption) {
		case Sixday:
			return 'Six day';

		case Everyday:
			return 'Every day';

		case EverydayPlus:
			return 'Every day package and digital subscription';

		case SixdayPlus:
			return 'Six day package and digital subscription';

		case WeekendPlus:
			return 'Weekend package and digital subscription';

		case SaturdayPlus:
			return 'Saturday package and digital subscription';

		case SundayPlus:
			return 'Sunday package and digital subscription';

		default:
			return productOption;
	}
};

export enum Channel {
	Observer = 'observer',
	ObserverAndGuardian = 'gardian_and_observer',
	Guardian = 'guardian',
}

export type LabelProps = {
	text: string;
	channel: Channel;
};
export const getLabel = (productOption: PaperProductOptions): LabelProps => {
	switch (productOption) {
		case 'Sunday':
			return {
				text: 'The Observer',
				channel: Channel.Observer,
			};
		case 'Weekend':
		case 'Everyday':
			return {
				text: 'The Guardian + the Observer',
				channel: Channel.ObserverAndGuardian,
			};
		default:
			return {
				text: 'The Guardian',
				channel: Channel.Guardian,
			};
	}
};
