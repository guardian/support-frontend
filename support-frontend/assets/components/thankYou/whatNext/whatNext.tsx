import { palette } from '@guardian/source/foundations';
import OrderedList from 'components/list/orderedList';
import type {
	ActiveProductKey,
	ActiveRatePlanKey,
} from 'helpers/productCatalog';
import type { ObserverPrint } from 'pages/paper-subscription-landing/helpers/products';
import {
	isGuardianWeeklyGiftProduct,
	isGuardianWeeklyProduct,
	isPrintProduct,
} from 'pages/supporter-plus-thank-you/components/thankYouHeader/utils/productMatchers';
import BulletPointedList from '../utilityComponents/BulletPointedList';
import { helpCenterCta, myAccountCta } from './whatNextCta';

export function WhatNext({
	productKey,
	ratePlanKey,
	amount,
	observerPrint,
	startDate,
	isSignedIn = false,
}: {
	productKey: ActiveProductKey;
	ratePlanKey: ActiveRatePlanKey;
	amount: string;
	startDate?: string;
	isSignedIn?: boolean;
	observerPrint?: ObserverPrint;
}) {
	const isSubscriptionCard = productKey === 'SubscriptionCard';
	const isGuardianAdLite = productKey === 'GuardianAdLite';
	const isGuardianWeekly = isGuardianWeeklyProduct(productKey);
	const isGuardianPrint = isPrintProduct(productKey) && !observerPrint;

	if (isGuardianWeekly) {
		const lookoutForEmailCopy =
			'Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.';
		const manageSubscriptionComponent = (
			<>
				You can manage your subscription by visiting {myAccountCta}. For any
				other queries please visit the {helpCenterCta('Help centre')}.
			</>
		);
		const guardianWeeklyItems = [
			lookoutForEmailCopy,
			'Your magazine will be delivered to your door. Please allow 1 to 7 days after publication date for your magazine to arrive, depending on national post services.',
			manageSubscriptionComponent,
		];
		const guardianWeeklyGiftItems = [
			lookoutForEmailCopy,
			'We’re unable to contact the gift recipient directly - make sure to let them know the gift is on its way',
			<>
				Each copy will be delivered to the gift recipient’s door. Here’s a
				reminder of {helpCenterCta('how home delivery works')}.
			</>,
			manageSubscriptionComponent,
		];

		const isGuardianWeeklyGift = isGuardianWeeklyGiftProduct(
			productKey,
			ratePlanKey,
		);
		return (
			<OrderedList
				items={
					isGuardianWeeklyGift ? guardianWeeklyGiftItems : guardianWeeklyItems
				}
			/>
		);
	}

	if (observerPrint ?? isGuardianPrint) {
		const observerItem =
			'The Observer team will be in touch shortly via email to welcome you. Check your spam folder or add the observer.co.uk domain to your preferred sender list.';

		const subscriptionCardItems = [
			'You should receive an email confirming the details of your subscription, and another email shortly afterwards that contains details of how you can pick up your newspapers from tomorrow.',
			'You will receive your Subscription Card in your subscriber pack in the post, along with your home delivery letter.',
			'Visit your chosen participating newsagent to pick up your newspaper using your Subscription Card, or arrange a home delivery using your delivery letter.',
		];

		const paperItems = [
			'Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.',
			'Your newspaper will be delivered to your door.',
		];

		const baseItems = isSubscriptionCard ? subscriptionCardItems : paperItems;
		const allItems = [...baseItems, observerPrint && observerItem];

		return <OrderedList items={allItems.filter(Boolean)} />;
	}

	if (isGuardianAdLite) {
		const guardianAdLiteItems = [
			'You will receive an email confirming the details of your subscription',
			startDate
				? `Your payment of £${amount}/month will be taken on ${startDate}`
				: '',
			isSignedIn
				? 'You can now start reading the Guardian website on all your devices without personalised advertising'
				: '',
		];

		return (
			<BulletPointedList
				items={guardianAdLiteItems.filter(Boolean)}
				color={palette.neutral[7]}
			/>
		);
	}

	return null;
}
