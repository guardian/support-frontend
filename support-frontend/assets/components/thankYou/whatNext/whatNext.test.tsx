import { render } from '@testing-library/react';
import { ObserverPrint } from 'pages/paper-subscription-landing/helpers/products';
import { WhatNext } from './whatNext';

describe('Summary Ts&Cs Snapshot comparison', () => {
	const whatNextItems = [
		[true, ObserverPrint.Paper],
		[false, ObserverPrint.SubscriptionCard],
	];
	it.each(whatNextItems)(
		`whatNext renders correctly (signedIn=%s, %s list)`,
		(signIn, observerPaperType) => {
			const { container } = render(
				<WhatNext
					amount={'12'}
					startDate={'Friday, March 28, 2025'}
					isSignedIn={signIn as boolean}
					observerPrint={observerPaperType as ObserverPrint}
					isGuardianWeekly={false}
					isGuardianPrint={false}
					isSubscriptionCard={
						observerPaperType === ObserverPrint.SubscriptionCard
					}
				/>,
			);
			expect(container).toMatchSnapshot();
		},
	);
});
