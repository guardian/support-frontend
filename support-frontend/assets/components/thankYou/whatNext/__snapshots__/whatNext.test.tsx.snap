// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Ts&Cs Snapshot comparison whatNext renders correctly (signedIn=false, ObserverSubscriptionCard list) 1`] = `
<div>
  <ol>
    <li
      class="css-k6etrx"
    >
      You should receive an email confirming the details of your subscription, and another email shortly afterwards that contains details of how you can pick up your newspapers from tomorrow.
    </li>
    <li
      class="css-k6etrx"
    >
      You will receive your Subscription Card in your subscriber pack in the post, along with your home delivery letter.
    </li>
    <li
      class="css-k6etrx"
    >
      Visit your chosen participating newsagent to pick up your newspaper using your Subscription Card, or arrange a home delivery using your delivery letter.
    </li>
    <li
      class="css-k6etrx"
    >
      The Observer team will be in touch shortly via email to welcome you. Check your spam folder or add the observer.co.uk domain to your preferred sender list.
    </li>
  </ol>
</div>
`;

exports[`Summary Ts&Cs Snapshot comparison whatNext renders correctly (signedIn=true, Homedelivery list) 1`] = `
<div>
  <ol>
    <li
      class="css-k6etrx"
    >
      Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.
    </li>
    <li
      class="css-k6etrx"
    >
      Your newspaper will be delivered to your door.
    </li>
    <li
      class="css-k6etrx"
    >
      The Observer team will be in touch shortly via email to welcome you. Check your spam folder or add the observer.co.uk domain to your preferred sender list.
    </li>
  </ol>
</div>
`;

exports[`Summary Ts&Cs Snapshot comparison whatNext renders correctly (signedIn=true, ObserverPaper list) 1`] = `
<div>
  <ol>
    <li
      class="css-k6etrx"
    >
      Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.
    </li>
    <li
      class="css-k6etrx"
    >
      Your newspaper will be delivered to your door.
    </li>
    <li
      class="css-k6etrx"
    >
      The Observer team will be in touch shortly via email to welcome you. Check your spam folder or add the observer.co.uk domain to your preferred sender list.
    </li>
  </ol>
</div>
`;
